app/static/src/app/store/plugins.ts

```diff
@@ -1,8 +1,18 @@
 import { MutationPayload, Store } from "vuex";
 import { AppState } from "./appState/state";
+import { StateUploadMutations } from "./appState/mutations";
 
 export const logMutations = (store: Store<AppState>): void => {
     store.subscribe((mutation: MutationPayload) => {
         console.log(mutation.type);
     });
 };
+
+export const persistState = (store: Store<AppState>): void => {
+    store.subscribe((mutation: MutationPayload) => {
+        if (!StateUploadMutations.includes(mutation.type) && !mutation.type.startsWith("errors")) {
```

Comment:
Thinking about performance, this will run too frequently, at the moment almost every action is queued. Could this be optimised to run only for major milestones ? 

Maybe when a user fits a model or run or anything major.

Reply:
We do want to retain all state changes, even quite minor ones (e.g. parameter changes, vis tab changes). Some state changes aren't going to be relevant (e.g. setting the solutions) but those will always be associated with changes we do want to persist (e.g. hasResult) so it didn't seem useful to specify those separately.

The idea with the queueing that that we won't actually send a save request for each mutation, but will batch up those that happen in quick succession (new queued interval id will supersede any waiting one) so the expensive part (the request) won't happen on every state change. It's the same pattern that we currently use in HINT.